validator/api/src/main/java/tech/pegasys/teku/validator/api/KeyStoreFilesLocator.java

```diff
@@ -80,12 +82,29 @@ private void parseEntry(final String keyFileName, final String passwordFileName)
               passwordFileName));
     }
     if (keyFile.isFile()) {
-      pathMap.putIfAbsent(keyFile.toPath(), passwordFile.toPath());
+      if (!isDepositDataFile(keyFile)) pathMap.putIfAbsent(keyFile.toPath(), passwordFile.toPath());
     } else {
       parseDirectory(keyFile, passwordFile);
     }
   }
 
+  private boolean isDepositDataFile(final File keyFile) {
```

Comment:
I'd probably filter by filename before checking the file contents. Deposit data files will always start with `deposit_data-`.